src/mlir/generator.cc

```diff
@@ -267,26 +267,41 @@ namespace mlir::verona
       return cacheReturn(getImm(context));
     else if (name == "bool")
       return cacheReturn(BoolType::get(context));
+    else if (name == "unk")
+      return cacheReturn(UnknownType::get(context));
 
     // Numeric
     size_t bitWidth = 0;
     name.substr(1).getAsInteger(10, bitWidth);
```

Comment:
Doesn't `getAsInteger` return false if it fails to parse?  Wouldn't it be better to use that to validate the width?

```C++
if (!substr(1).getAsInteger(10,bitWidth))
  bitWidth = 64;
```
Or if you envisage not having default as always `64`, then you could do
```C++
bool validWidth = substr(1).getAsInteger(10,bitWidth);
assert((!validWidth) || (bitWidth >= 8 && bitWidth <=128));
```

Reply:
Interesting. I'm not fully happy with having a default, but this looks like a cleaner implementation.